hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Retries.java

```diff
@@ -26,11 +26,23 @@
 import org.apache.hadoop.classification.InterfaceStability;
 
 /**
- * Declaration of retry policy for documentation only.
- * This is purely for visibility in source and is currently package-scoped.
- * Compare with {@link org.apache.hadoop.io.retry.AtMostOnce}
- * and {@link org.apache.hadoop.io.retry.Idempotent}; these are real
- * markers used by Hadoop RPC.
+ * <p>
+ *   Annotations to inform the caller of an annotated method whether
```

Comment:
I tried to improve the documentation of this class here.  I'm curious if I have captured the real intended meaning/purpose of these annotations?  If so, hopefully it will help the next developer understand the goal better than I did initially.

Reply:
Yes, though I'd be stricter about the wrapping of retries. You mustn't retry around a retrying operation as you'll just cause things to take so long that operations will time out anyway. The raw/translated flags are markers about whether to expect AWS-library Runtime Exceptions or translated stuff, or a mixture. It's less critical, but once everything is pure IOE, there's no need to catch and convert again.

Reply:
Tweaked in https://github.com/apache/hadoop/pull/666/commits/ef91a67b3a10d12817e0bf38c47c1b1403084013